### ISSUE TITLE
Wrong handling of return type in macros

### ISSUE THREAD

**Issue by micheleCTDE** [uncrustify/uncrustify#3852]
uncrustify master (commit 765a60d0).
Config file as per attachment.

Input file
```
#define DEFINE_SYMBOL(ret, fn, ...)   ret (*OVERRIDE_SYMBOL(fn))(__VA_ARGS__)
```
Output
```
#define DEFINE_SYMBOL(ret, fn, ...) ret(*OVERRIDE_SYMBOL(fn))(__VA_ARGS__)
```
Note that the space between `ret` and `(` has been removed. This happens because `ret` is not considered a `type` but a function call due to the additional parenthesis for `(fn)`. 
```
#   1> FUNC_CALL| NONE|PARENT_NOT_SET[ 37/ 39/ 42/  3][1/1/0][ 20 0001][0-0] ret
```
The following input line 
```
#define DEFINE_SYMBOL(ret, fn, ...)   ret (*fn)(__VA_ARGS__)
```
would instead be handled correctly since there are no more parenthesis after the `*` and `)`.

Will have a PR ready soon.

[c.cfg.zip](https://github.com/uncrustify/uncrustify/files/9645387/c.cfg.zip)

**Comment by micheleCTDE** [comment]
The problem can also be reproduce with a simple config file as follow:
```
sp_func_call_paren = remove
```

**Comment by micheleCTDE** [comment]
This was solved by PR #3853
